common/src/main/java/org/opensearch/ml/common/conversational/ConversationMeta.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2023 Aryn
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.opensearch.ml.common.conversational;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Map;
+
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.search.SearchHit;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+/**
+ * Class for holding conversational metadata
+ */
+@AllArgsConstructor
+public final class ConversationMeta implements Writeable, ToXContentObject {
+
+    @Getter
+    private String id;
+    @Getter
+    private Instant created;
+    @Getter
+    private Instant lastHit;
+    @Getter
+    private int numInteractions;
```

Comment:
One conversation can have multiple interactions? Can you explain the definition of "interaction"? I guess it may mean: human send out one message to model, then model return a message back, these two messages will form one interaction, right?

Sorry this is a huge PR, I need some time to go though all. I just put my question and understanding here as early as possible. Apologize if that's a stupid question as you have mentioned this somewhere in this PR.

Reply:
Yep, you've got it! A conversation is a list of interactions; an interaction is an input/output pair of messages.